site/zenodo_rdm/legacy/serializers/schemas/legacyjson.py

```diff
@@ -139,9 +139,8 @@ def dump_files(self, obj):
         bucket_url = obj["links"].get("bucket")
         files_url = obj["links"].get("files")
         for key, f in obj["files"].get("entries", {}).items():
-            file_id = f["id"]
             if files_url:
-                links = {"self": f"{files_url}/{file_id}"}
+                links = {"self": f"{files_url}/{key}"}
```

Comment:
that was producing wrong link i.e `api/records/<recid>/files/<file_id>`

Reply:
We have to be careful, some of these links are important for partners (like LORY).

Reply:
if they had hardcode bucket url, I checked and it still works but the generated links changed to working ones.